common/canonical_serialization/src/lib.rs

```diff
@@ -560,3 +596,43 @@ impl CanonicalDeserialize for BTreeMap<Vec<u8>, Vec<u8>> {
         Ok(deserializer.decode_btreemap()?)
     }
 }
+
+impl CanonicalSerialize for &str {
+    fn serialize(&self, serializer: &mut impl CanonicalSerializer) -> Result<()> {
+        serializer.encode_string(self)?;
+        Ok(())
+    }
+}
+
+impl CanonicalDeserialize for String {
```

Comment:
It sounds like a macro using a `stringify!($name)` on the struct type would help here.

Reply:
Will look into that and if it is straightforward enough will do it. Otherwise can add it to another issue.